_posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md

```diff
@@ -0,0 +1,80 @@
+
+The thing is, especially in the software consulting line of business, doubling down on just one thing won't get you very far. It might not even carry you through the first job interview unless you are applying for very entry-level work with some training coming up immediately. That's because we don't know what skills will be needed for the next project. If you are only a master of one skill, it might be more difficult to find yourself a good project to work with, since others in the team would need to carry all other areas and aspects of the project, until you have learned them. It gives a much better base for consulting work to be good with many parts of the software craft (as opposed to being a master of one).
+
+So, while specialization is still rather a valid thing to do, you should also invest in learning a wide array of skills for software crafting, as listed above. These days you definitely need to understand automated testing, delivery, security, and cloud platforms, in addition to Frontend, Backend, and databases. But you can still specialize, you can double on on one or two things that bring you pleasure. You can actively practice them, learn more about them, and be the go-to-person for said techniques.
+
+But there's also another route. I'm a Java coder. That's been my forte. I started it in the 1990s, then I did a long period of teaching it for money, which meant I needed to first master it myself, and keep up to date. I would say Java is something I can code in my sleep, and sometimes do, so that's definitely my specialization and strength. But would you believe it if I said I have barely used any Java for several years now? I've been relying on my 25 years of experience on all things, and my ability to rapidly pick up new things based on that. If I drew my skills profile right now it would be crazy. I can do many things. Some might argue even any things. My recent years have been filled for example by Splunk and Excel coding, a lot of Python, Scala, some Kotlin, Gitlab pipelines, and lots and lots of cloud acronyms. Did you know that AWS has hundreds of services and thousands of acronyms? And each one is a specialization in itself.
+
+So I would call myself a generalist. That's been my specialization. Meaning that I actively pursue that and keep my generalization up to date. I know few things really in-depth, but I know a hundred things on a level that I can rapidly refresh up to today's requirements if there's a need. A few years back someone asked me about XSLT templates, so I whipped up from my ancient technologies archive a nice recursion template that lets you do crazy stuff with minimal code. That gives me much flexibility to jump in many places. Unfortunately, that comes with a downside too. A constant visitor.
```

Comment:
"whipped up" means "create", "whipped out" is probably more  appropriate here. Also, yay for XSLT!

"That gives" -- "That" refers to the XSLT snippet. Do you mean to refer to the fact that you're a generalist? Perhaps "My generalism gives..."